### PR TITLE
fix: codegen - project paths with spaces

### PR DESCRIPTION
**Pull request by mysport12**
- Fixed iOS codegen script incorrectly splitting root project paths that contain spaces

## Summary

https://github.com/react-native-community/releases/issues/214#issuecomment-793089063

iOS builds were failing on 0.64.0-rc.4 for projects that contained spaces in the root directory path. The error logs pointed to the codegen script not being able to find a directory. The path was being split at a space in one of the folder names. This PR modifies the codegen script to include the spaces and use the entire project root path.

## Changelog

fix: codegen script failing for iOS builds on projects with spaces in root directory path

## Test Plan

Failing Test: Upgrade or init a new project and make sure that the project root directory contains a space (ex: /Users/test/cool projects/app/). With a clean install of node_modules and pods, attempt to build the project with Xcode. The build fails with an error running the script in FBReactNativeSpec (no such file or directory).

Passing Test: Include the changes presented in this PR and rerun the failing test (clean node_modules + PR patch/clean pods). The app should build.

### CODE REVIEW

**Comment by mlazari** [comment]
I think ideally there shouldn't be any absolute paths if possible, as it makes Podfile.lock change depending on the path where the project is. See https://github.com/facebook/react-native/issues/31121

**Comment by mlazari** [comment]
I didn't try it, but I think this PR might fix both issues: https://github.com/facebook/react-native/pull/31195

File: scripts/react_native_pods.rb [hunk 1]
```diff
@@ -192,10 +192,10 @@ def use_react_native_codegen!(spec, options={})
...
     :input_files => [srcs_dir],
     :output_files => ["$(DERIVED_FILE_DIR)/codegen-#{codegen_modules_library_name}.log"].concat(generated_files),
-    :script => "set -o pipefail\n\nbash -l -c '#{env_vars} CODEGEN_MODULES_LIBRARY_NAME=#{codegen_modules_library_name} #{File.join(__dir__, "generate-specs.sh")}' 2>&1 | tee \"${SCRIPT_OUTPUT_FILE_0}\"",
+    :script => "set -o pipefail\n\nbash -l -c \"#{env_vars} CODEGEN_MODULES_LIBRARY_NAME='#{codegen_modules_library_name}' '#{File.join(__dir__, "generate-specs.sh")}'\" 2>&1 | tee \"${SCRIPT_OUTPUT_FILE_0}\"",
     :execution_position => :before_compile,
     :show_env_vars_in_log => true
   }
 
-  spec.prepare_command = "#{mkdir_command} && touch #{generated_files.reduce() { |str, file| str + " " + file }}"
+  spec.prepare_command = "#{mkdir_command} && touch #{generated_files.reduce() { |str, file| 'str' + " " + 'file' }}"
```
Comment (rrrasti) [inline scripts/react_native_pods.rb:200]: This should be changed.

```suggestion
  spec.prepare_command = "#{mkdir_command} && touch #{generated_files.map {|file| "'#{file}'"}.join(" ")}"
```

**Comment by rrrasti** [comment]
This PR might fix #31181 and #31198.

**Comment by rb185264** [comment]
This issue is still happening on 0.69

**Pull request closed without merging**